### PR TITLE
Create load_image helper function

### PR DESCRIPTION
**Pull request by capjamesg**
This PR implements a new `load_image` helper function. This function accepts either a file name, a `cv2` image, or a `PIL` image. A consuming application can specify a custom output.

This PR works toward addressing #56.

Consider a scenario where a user is working with `autodistill_clip` and already has an image in memory. Right now, `.predict()` will open an image. With this PR, and changes to all base models, `.predict()` could instead take the required NumPy array directly, instead of opening the file again.

### CODE REVIEW

**Comment by capjamesg** [comment]
This PR also unblocks #86.

**Review by yeldarby — APPROVED** [review]
Noting discussion points to change from our call:
* Don't convert to cv2_image until just before the `return` since it's often unused
* Be able to accept cv2 images

Discussed but ruled out:
* Reuse [the image type inference logic from `inference`](https://github.com/roboflow/inference/blob/c486d2c74dc783c8ad4809a412b012c1748828af/inference/core/utils/image_utils.py#L122) because the use-case is different & that logic is too complicated (`autodistill` is used in a much narrower context)

File: autodistill/helpers.py [hunk 1]
```diff
@@ -1,14 +1,58 @@
...
+    image: Any,
+    return_format="cv2",
+) -> Any:
+    if return_format not in ACCEPTED_RETURN_FORMATS:
+        raise ValueError(f"return_format must be one of {ACCEPTED_RETURN_FORMATS}")
+
+    if isinstance(image, Image.Image) and return_format == "PIL":
+        return image
+    elif isinstance(image, Image.Image) and return_format == "cv2":
+        return np.array(image)
```
Comment (yeldarby) [inline autodistill/helpers.py:29]: I think with PIL to cv2 you have to convert the channels from RGB to BGR? (And vice versa for cv2 to others)

Comment (capjamesg) [reply autodistill/helpers.py]: I have added the requisite conversion code in my latest commit.

**Pull request merged**